Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -11845,6 +11912,32 @@ def comma_separated_mapping_to_dict(raw_text):
     return mapping_dict
 
 
+class ManagedSleep:
+    def __init__(self,):
+        """
+        Initializes the sleeper with the given TTL in seconds.
+
+        """
+        self.run_duration = demisto.callingContext.get('context', {}).get('runDuration', 5) * 60
```

Comment:
Does it matter? we have a default value set to be 5 (as discussed with Liad)

Reply:
if the user passes a custom execution-timeout of 10 minutes, and there's no `runDuration` in this XSOAR version, for example if we use the SafeSleep for 6 minutes it will fail although it shouldn't. an edge case, but possible. correct me if I'm wrong

Reply:
@dantavori 
I suggested:
* Not to throw an exception, TTL management shouldn't be delegated to the managed sleep IMO. It's not its job
* Instead of throwing an exception, log a warning + cap the sleep duration since we're passed what's expected
* Manage a scripts TTL by using a "managed main" wrapper instead

You're completely right about the user param.

BTW - another option is to add a condition -> if we managed to retrieve the value from the platform act ... other wise act differently.